### PR TITLE
fix(publish): --config.unknown CLI options needs special handling when passing them to npm 

### PR DESCRIPTION
**Pull request by naruaway**
This should fix https://github.com/pnpm/pnpm/issues/9084

If this direction looks good, I can add tests

So far I quickly verified by modifying `dist/pnpm.cjs` directly inside `node_modules` for the corresponding place

### CODE REVIEW

**Comment by zkochan** [comment]
Yes, this is probably OK.

File: releasing/plugin-commands-publishing/src/publish.ts [hunk 1]
```diff
@@ -205,6 +205,8 @@ Do you want to continue?`,
       args.splice(index, 2)
     }
   }
+  // Convert Pnpm's "unknown config" form into regular form so that npm can understand it
+  args = args.map(a => a.replace(/^--config\./, '--'))
```
Comment (naruaway) [inline releasing/plugin-commands-publishing/src/publish.ts:209]: Note: I just realized that we need to make sure we should not apply this conversion after `--`. I'll fix later after getting some signal from maintainers

**Comment by naruaway** [comment]
@zkochan Thanks for really quick response! I'll make sure handle edge cases and will add tests later today

**Comment by naruaway** [comment]
After some consideration, now I would prefer https://github.com/pnpm/pnpm/pull/9089 over this PR. Closing for now
